sync/src/relayer/block_transactions_process.rs

```diff
@@ -1,21 +1,33 @@
+use crate::block_status::BlockStatus;
 use crate::relayer::block_transactions_verifier::BlockTransactionsVerifier;
+use crate::relayer::error::{Error, Misbehavior};
+use crate::relayer::types::BlockTransactions;
 use crate::relayer::Relayer;
-use ckb_core::transaction::Transaction;
 use ckb_network::{CKBProtocolContext, PeerIndex};
-use ckb_protocol::{cast, BlockTransactions, FlatbuffersVectorIterator};
+use ckb_protocol::BlockTransactions as FbsBlockTransactions;
+use ckb_protocol::RelayMessage;
 use failure::Error as FailureError;
+use flatbuffers::FlatBufferBuilder;
 use numext_fixed_hash::H256;
 use std::collections::hash_map::Entry;
 use std::convert::TryInto;
 use std::sync::Arc;
 
+// Keeping in mind that short transaction IDs are expected to occasionally collide.
+// On receiving block-transactions message, if reconstruct the block with a different transactions_root,
+// - if all the short_ids are included,
+// the node should ban the peer but not mark the block invalid
```

Comment:
I think the node should not ban the peer, unless the downloaded block is really invalid.

Reply:
Do you mean "if all the transactions are prefilled," here like the comment below?

Reply:
If the BlockTransactions includes all the transactions matched `short_ids` in the compact block. 
I need to update the comment to make it clear.

Reply:
update the comment, pls review again, thanks.